airsonic-main/src/main/java/org/airsonic/player/ajax/PlayQueueService.java

```diff
@@ -677,12 +739,53 @@ private PlayQueueInfo convert(HttpServletRequest request, Player player, boolean
                     formatFileSize(file.getFileSize(), locale), starred, albumUrl, streamUrl, remoteStreamUrl,
                     coverArtUrl, remoteCoverArtUrl));
         }
-        boolean isStopEnabled = playQueue.getStatus() == PlayQueue.Status.PLAYING && !player.isExternalWithPlaylist();
 
-        float gain = 0.0f;
-        gain = jukeboxService.getGain(player);
+        return entries;
+    }
+
+    private List<PlayQueueInfo.Entry> convertInternetRadio(HttpServletRequest request, Player player) throws Exception {
 
-        return new PlayQueueInfo(entries, isStopEnabled, playQueue.isRepeatEnabled(), playQueue.isRadioEnabled(), serverSidePlaylist, gain);
+        PlayQueue playQueue = player.getPlayQueue();
+        InternetRadio radio = playQueue.getInternetRadio();
+
+        final String radioHomepageUrl = radio.getHomepageUrl();
+        final String radioName = radio.getName();
+
+        List<PlayQueueInfo.Entry> entries = new ArrayList<>();
+        for (InternetRadioSource streamSource: internetRadioService.getInternetRadioSources(radio)) {
+            // Fake stream title using the URL
+            String streamUrl = streamSource.getStreamUrl();
+            String streamTitle = streamUrl;
+            String streamAlbum = radioName;
+            String streamGenre = "Internet Radio";
+            // Fake entry id so that the source can be selected
+            Integer streamId = -(1+entries.size());
+            Integer streamTrackNumber = entries.size();
+            Integer streamYear = 0;
+            entries.add(new PlayQueueInfo.Entry(
+                    streamId,          // Entry id
+                    streamTrackNumber, // Track number
+                    streamTitle,       // Use URL as stream title
```

Comment:
Since there is a comment here, I'm not convinced that it's worth having a temporary variable, is it?

Reply:
Yup, you're right. Tell me if the new commit sounds okay with you.